app/src/main/java/ee/ioc/phon/android/speechutils/AbstractAudioRecorder.java

```diff
@@ -182,7 +257,15 @@ protected void setState(State state) {
      */
     public synchronized byte[] consumeRecording() {
         byte[] bytes = getCurrentRecording(mConsumedLength);
+        // this is to avoid race (set the consumed length to be the recorded length though
+        // the last recording was empty while recorded length moved on - thus we always miss
+        // a part of the recording)
+        if (bytes == null)
+            return null;
+
         mConsumedLength = mRecordedLength;
+        if (mAlwaysListen)
```

Comment:
in case of continuous, once consumeRecording was called, all of the relevant buffer is consumed (by definition) and this means that at the end of the consumption process, the consume & recording sessions are the same